docs/testing.md

```diff
@@ -0,0 +1,179 @@
+---
+id: testing
+title: Testing
+---
+
+As your codebase expands, small errors and edge cases you don’t expect can cascade into larger failures. Bugs lead to bad user experience and ultimately, business losses. One way to prevent fragile programming is to test your code before releasing it into the wild. In this guide, we will cover different ways to ensure your app works as expected, ranging from static analysis to End to End tests.
+
+## Static Analysis
+
+The first step to improve your code quality is to start using static analysis tools. Static analysis checks your code for errors as you write it, but without running any of that code.
+
+- **Linters** analyze code to catch common errors, such as unused code, help avoid pitfalls or flag style guide no-nos like using tabs instead of spaces (or vice versa, depending on your configuration). Linters are configured by a set of rules - and your code needs to follow them.
+- **Type checking** ensures that the construct you’re passing to a function matches what the function was designed to accept, preventing passing a string to a counting function that expects an integer, for instance.
+
+React Native comes with two such tools configured out of the box: [ESLint](https://eslint.org/) for linting and [Flow](https://flow.org/en/docs/) for type checking. You can also use [TypeScript](https://www.typescriptlang.org/), which is a typed language that compiles to plain JavaScript.
+
+## Writing Testable Code
+
+To start with tests, you first need to write code that is testable. Consider a car’s engine. For some reason it won’t start. In older cars, figuring out what is wrong with the engine requires the mechanic to take the engine apart, piece by piece, until they can figure out the issue. In modern cars, engines are outfitted with sensors that keep an eye on parts of the system, determining the exact location of part failures—sometimes before the engine even breaks down! Sensors changed the way engines were designed and manufactured.
+
+Similarly, with software, writing your code in many small functions that you can test (like adding sensors to the different components of an engine) is more effective than writing your entire program in one file with many lines of code (where you can only test if the engine starts or not). In this way, writing testable code is intertwined with writing clean, modular code. We encourage you to explore this topic in other learning resources.
+
+To help make your app more testable, it’s a good idea to separate the view part of your app, React components, from business logic and app state (regardless of whether you use Redux, MobX or other solutions). This way, you can separate tests of your business logic — which shouldn’t rely on your React components — from the components themselves, whose job is primarily rendering your app’s UI!
```

Comment:
Is there another post online somewhere that we like that we can link to for how to write testable React / JS code?

Reply:
I did some googling, but didn't find too much content. Would you like to add some of these?

https://www.toptal.com/javascript/writing-testable-code-in-javascript  
https://testing.googleblog.com/2008/08/by-miko-hevery-so-you-decided-to.html?m=1 
https://www.toptal.com/qa/how-to-write-testable-code-and-why-it-matters 
https://dzone.com/articles/write-testable-code

Reply:
I endorse you writing your own post... with us :3